news/tests/test_update_user_task.py

```diff
@@ -0,0 +1,269 @@
+import json
+
+from django.core.cache import cache
+from django.test import TestCase, RequestFactory
+
+from basket import errors
+from mock import patch, ANY
+from ratelimit.exceptions import Ratelimited
+
+from news.utils import SUBSCRIBE, UNSUBSCRIBE, SET
+from news.views import update_user_task
+
+
+class UpdateUserTaskTests(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+
+        patcher = patch('news.views.upsert_contact')
+        self.upsert_contact = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        patcher = patch('news.views.upsert_user')
+        self.upsert_user = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        cache.clear()
```

Comment:
Yes. Need to not use dummy cache so that we can test rate limits, but they get in the way on certain other tests. Could mock instead but this seems cleaner.